backoffice/views.py

```diff
@@ -404,16 +404,21 @@ def news_list(request):
 @group_required('fun_backoffice')
 def news_detail(request, news_id=None):
     if news_id:
-        article = Article.objects.get(id=news_id)
+        search_query = {
+            'id': news_id
+        }
         if settings.FEATURES['USE_MICROSITES']:
-            assert article.microsite == microsite.get_value('SITE_NAME')
+            search_query['microsite'] = microsite.get_value('SITE_NAME')
+        article = get_object_or_404(Article, **search_query)
     else:
         article = None
 
-    form = ArticleForm(data=request.POST or None, instance=article)
-    if form.is_valid():
-        form.save()
-        return redirect('backoffice:news-list')
+    if request.method == 'POST':
+        form = ArticleForm(data=request.POST, files=request.FILES, instance=article)
+        if form.is_valid():
+            form.save()
+    else:
+        form = ArticleForm(instance=article)
 
```

Comment:
Ici on fait deux choses de manière différente :

1. On renvoie la page d'édition de l'article à la place de la page de listing des articles ; c'est un changement de comportement souhaité par la CA et qui leur permet de modifier plusieurs fois d'affilée un article tout en mettant à jour la prévisualisation.

2. On instancie un `form` en passant au constructeur `request.FILES`, ce qui est nécessité par l'upload de vignette. On aurait pu tout aussi bien instancier l'objet ainsi : `form = ArticleForm(data=request.POST or None, files=request.FILES or None, instance=article)`. Il m'a semblé que cette manière de faire était moins "standard" : https://docs.djangoproject.com/en/1.9/topics/forms/#the-view Ceci dit, c'est une question d'appréciation. Je considère qu'on a besoin de distinguer plus explicitement les cas en fonction du type de requête, mais je ne suis pas braqué là-dessus.